vm/actor/tests/deadline_test.rs

```diff
@@ -0,0 +1,671 @@
+};
+use bitfield::BitField;
+use clock::ChainEpoch;
+
+fn assert_deadline_info(
+    current: ChainEpoch,
+    period_start: ChainEpoch,
+    index: usize,
+    expected_deadline_open: ChainEpoch,
+) -> DeadlineInfo {
+    let di = DeadlineInfo {
+        current_epoch: current,
+        period_start,
+        index,
+        open: expected_deadline_open,
+        close: expected_deadline_open + WPOST_CHALLENGE_WINDOW,
+        challenge: expected_deadline_open - WPOST_CHALLENGE_LOOKBACK,
+        fault_cutoff: expected_deadline_open - FAULT_DECLARATION_CUTOFF,
+    };
+    let actual = compute_proving_period_deadline(period_start, current);
+    assert!(actual.period_started());
+    assert!(actual.is_open());
+    assert!(!actual.has_elapsed());
+    assert_eq!(di, actual);
+    return di;
+}
+
+#[test]
+fn pre_open_deadlines_test() {
+    // Current is before the period opens.
+    let curr: ChainEpoch = 0;
+    let mut period_start = FAULT_DECLARATION_CUTOFF + 1;
+    let mut di = compute_proving_period_deadline(period_start, curr);
+    assert_eq!(0, di.index);
+    assert_eq!(period_start, di.open);
+    assert!(!di.period_started());
+    assert!(!di.is_open());
+    assert!(!di.has_elapsed());
+    assert!(
+        !di.fault_cutoff_passed(),
+        format!(
+            "curr epoch: {} >= faultcutoff: {}",
+            di.current_epoch, di.fault_cutoff
+        )
+    );
+    assert_eq!(period_start + WPOST_PROVING_PERIOD - 1, di.period_end());
+    assert_eq!(period_start + WPOST_PROVING_PERIOD, di.next_period_start());
+    period_start = FAULT_DECLARATION_CUTOFF - 1;
+    di = compute_proving_period_deadline(period_start, curr);
+    assert!(di.fault_cutoff_passed());
+}
+
+#[test]
+fn offset_zero_test() {
+    let first_period_start: ChainEpoch = 0;
+
+    let mut di = assert_deadline_info(0, first_period_start, 0, 0);
+    assert_eq!(-WPOST_CHALLENGE_LOOKBACK, di.challenge);
+    assert_eq!(-FAULT_DECLARATION_CUTOFF, di.fault_cutoff);
+    assert!(di.is_open());
+    assert!(di.fault_cutoff_passed());
+
+    assert_deadline_info(1, first_period_start, 0, 0);
+    assert_deadline_info(WPOST_CHALLENGE_WINDOW - 1, first_period_start, 0, 0);
+    assert_deadline_info(
+        WPOST_CHALLENGE_WINDOW,
+        first_period_start,
+        1,
+        WPOST_CHALLENGE_WINDOW,
+    );
+    assert_deadline_info(
+        WPOST_CHALLENGE_WINDOW,
+        first_period_start,
+        1,
+        WPOST_CHALLENGE_WINDOW,
+    );
+    assert_deadline_info(
+        WPOST_CHALLENGE_WINDOW * 2 - 1,
+        first_period_start,
+        1,
+        WPOST_CHALLENGE_WINDOW,
+    );
+    assert_deadline_info(
+        WPOST_CHALLENGE_WINDOW * 2,
+        first_period_start,
+        2,
+        WPOST_CHALLENGE_WINDOW * 2,
+    );
+    assert_deadline_info(
+        WPOST_PROVING_PERIOD - 1,
+        first_period_start,
+        WPOST_PERIOD_DEADLINES - 1,
+        WPOST_PROVING_PERIOD - WPOST_CHALLENGE_WINDOW,
```

Comment:
idk definitely optional, but could alias these constants imports to make all this easier to follow throughout the file. Would make it easier to track and updates changes later